pkg/api/customization/project/project_actions.go

```diff
@@ -0,0 +1,147 @@
+package project
+
+import (
+	"fmt"
+
+	"net/http"
+
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/parse"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/client/management/v3"
+)
+
+func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+	resource.AddAction(apiContext, "setpodsecuritypolicytemplate")
+}
+
+type Handler struct {
+	ProjectClient v3.ProjectInterface
+}
+
+func (h *Handler) Actions(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	switch actionName {
+	case "setpodsecuritypolicytemplate":
+		return h.setPodSecurityPolicyTemplate(actionName, action, apiContext)
+	}
+
+	return errors.Errorf("bad action %v", actionName)
+}
+
+func (h *Handler) setPodSecurityPolicyTemplate(actionName string, action *types.Action, request *types.APIContext) error {
+	actionInput, err := parse.ReadBody(request.Request)
+	if err != nil {
+		return err
+	}
+
+	schema := request.Schemas.Schema(&managementschema.Version, client.PodSecurityPolicyTemplateProjectBindingType)
+	if schema == nil {
+		return fmt.Errorf("no %v store available", client.PodSecurityPolicyTemplateProjectBindingType)
+	}
+
+	binding, err := h.createOrUpdateBinding(request, schema, actionInput)
+	if err != nil {
+		return err
+	}
+
+	err = h.updateProjectPSPTID(request, actionInput)
+	if err != nil {
+		return err
+	}
+
+	request.WriteResponse(http.StatusOK, binding)
+
+	return nil
+}
+
+func (h *Handler) createOrUpdateBinding(request *types.APIContext, schema *types.Schema,
+	actionInput map[string]interface{}) (map[string]interface{}, error) {
+	bindings, err := schema.Store.List(request, request.Schema, &types.QueryOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var binding map[string]interface{}
+	for _, candidate := range bindings {
+		if candidate["projectId"] == request.ID {
+			binding = candidate
+			break
+		}
+	}
+
+	if binding == nil {
+		binding, err = h.createNewBinding(request, schema, actionInput)
+		if err != nil {
+			return nil, fmt.Errorf("error creating binding: %v", err)
+		}
+	} else {
+		binding, err = h.updateBinding(binding, request, schema, actionInput)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return binding, nil
+}
+
+func (h *Handler) updateProjectPSPTID(request *types.APIContext, actionInput map[string]interface{}) error {
+	project, err := request.Schema.Store.ByID(request, request.Schema, request.ID)
+	if err != nil {
+		return fmt.Errorf("error getting current project: %v", err)
+	}
+
+	status, ok := project["status"]
```

Comment:
This logic is incorrect. I put a breakpoint here and status is not in return resource. Here's why:
When you use a store from a schema, you are getting a presentation of the resource that has been transformed for our API. Our API does not directly expose the status field. It translate the fields in it to top-level read-only fields.
To update this field on status, you'll probably need to use the real k8s client, not the store.